src/Compiler/Interactive/fsi.fs

```diff
@@ -1329,23 +1329,34 @@ let internal mkBoundValueTypedImpl tcGlobals m moduleName name ty =
     entity, v, CheckedImplFile.CheckedImplFile(qname, [], mty, contents, false, false, StampMap.Empty, Map.empty)
 
 
-let scriptingDirectory =
-    let createDirectory directory =
+let scriptingSymbolsPath =
+    let createDirectory path =
         lazy
             try
-                if not (Directory.Exists(directory)) then
-                    Directory.CreateDirectory(directory) |> ignore
+                if not (Directory.Exists(path)) then
+                    Directory.CreateDirectory(path) |> ignore
 
-                directory
+                path
             with _ ->
-                directory
+                path
 
     createDirectory (Path.Combine(Path.GetTempPath(), $"{DateTime.Now:s}-{Guid.NewGuid():n}".Replace(':', '-')))
 
+let deleteScriptingSymbols () =
+    try
+#if !DEBUG
+            if scriptingSymbolsPath.IsValueCreated then
+                if Directory.Exists(scriptingSymbolsPath.Value) then
+                    Directory.Delete(scriptingSymbolsPath.Value, true)
+#else
+            ()
+#endif
+    with _ ->
+        ()
 
+AppDomain.CurrentDomain.ProcessExit |> Event.add (fun _ -> deleteScriptingSymbols ())
```

Comment:
What if FSI is used as an API?

Consumption of this module will still mean adding this event and never losing it, right?

Reply:
Yes.

Reply:
It's a ProcessExit event ... by definition it needs to hang around forever.